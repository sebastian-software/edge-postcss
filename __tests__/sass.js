import { compile } from "./core"

// ====================================================
// ================ SASS INSPIRED =====================
// ====================================================

test("Config Maps", () =>
  compile("h1 { background: map(Config, bgcolor); }")
)

test("Sassy Mixins", () =>
  compile("@mixin simple{ color: red; } h1 { @include simple; }")
)

test("Sassy Variables", () =>
  compile("$bgColor: red; h1 { background: $bgColor; }")
)

test("Nested Basic", () =>
  compile("body { h1 { font-weight: bold; } h2 { font-weight: normal; }}")
)

test("Nested Parent Selector", () =>
  compile("body { ul { li { &:first-child { margin-top: 0; }}}}")
)
