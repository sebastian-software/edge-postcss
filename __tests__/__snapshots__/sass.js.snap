// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Maps 1`] = `
"h1{
background:darkyellow
}"
`;

exports[`Nested Basic 1`] = `
"body h1{
font-weight:700
}

body h2{
font-weight:400
}"
`;

exports[`Nested Parent Selector 1`] = `
"body ul li:first-child{
margin-top:0
}"
`;

exports[`Sassy Mixins 1`] = `
"h1{
color:#ff4136
}"
`;

exports[`Sassy Variables 1`] = `
"h1{
background:#ff4136
}"
`;
